syft/generic/object_storage.py

```diff
@@ -79,35 +81,40 @@ def set_obj(self, obj: Union[FrameworkTensorType, AbstractTensor]) -> None:
             obj: A torch or syft tensor with an id.
         """
         self._objects[obj.id] = obj
-
-    def rm_obj(self, remote_key: Union[str, int]):
+        # Add entry in the tag index
+        if obj.tags:
+            for tag in obj.tags:
+                if tag not in self._tag_to_object_ids:
+                    self._tag_to_object_ids[tag] = {obj.id}
+                else:
+                    self._tag_to_object_ids[tag].add(obj.id)
+
+    def rm_obj(self, remote_key: Union[str, int], force=False):
         """Removes an object.
 
         Remove the object from the permanent object registry if it exists.
 
         Args:
             remote_key: A string or integer representing id of the object to be
```

Comment:
Why is the argument called remote_key?

Reply:
No clue! it was changed at some poitn by someone :D